Tests/NIOTests/BootstrapTest.swift

```diff
@@ -529,6 +529,25 @@ class BootstrapTest: XCTestCase {
         XCTAssertNil(NIOPipeBootstrap(validatingGroup: elg))
         XCTAssertNil(NIOPipeBootstrap(validatingGroup: el))
     }
+    
+    func testShorthandOptionsAreEquivalent() throws {
+        func bindAndGetReuseAddrOption(_ applyBootstrapOptions : (ServerBootstrap) -> ServerBootstrap) throws ->
+                ChannelOptions.Types.SocketOption.Value {
+            let bootstrap = applyBootstrapOptions(ServerBootstrap(group: group))
+            let serverChannel = try bootstrap.bind(host: "127.0.0.1", port: 0).wait()
+            let reuseValue = try serverChannel.getOption(ChannelOptions.socketOption(.reuseaddr)).wait()
+            try serverChannel.close().wait()
+            return reuseValue
+        }
+        
+        let sbLongReuseValue = try bindAndGetReuseAddrOption { bs in
+            bs.serverChannelOption(ChannelOptions.socketOption(.reuseaddr), value: 1) }
```

Comment:
whilst we're here we should also test the equivalence to the traditional socket options (`ChannelOptions.socket(.init(SOL_SOCKET), .init(SO_REUSEPORT), value: 1`)

Reply:
Reuse port is not one of the currently supported shorthands.  You are right that not all options are tested though, nor all bootstraps yet.

Reply:
@PeterAdams-A sorry, yes, typed `REUSEPORT` instead of `REUSEADDR`, no need to make `REUSEPORT` convenient but we should test the now three way to set `SO_REUSEADDR`:

- `.socket(.init(SOL_SOCKET), .init(SO_REUSEADDR), value: 1)`
- `.socketOption(.reuseaddr), value: 1)`
- `.allowImmediateLocalEndpointAddressReuse`

and make sure they all do the same

Reply:
I think I've added traditional api to the mix for this option.